geo/src/algorithm/affine_ops.rs

```diff
@@ -453,6 +453,34 @@ impl<U: CoordFloat> AffineTransform<U> {
     }
 }
 
+impl Index<&str> for AffineTransform {
+    type Output = f64;
+
+    /// Get elements from the AffineTransform matrix using string-based indexing.
+    ///
+    /// ```
+    /// let transform = AffineTransform::new(10.0, 0.0, 400_000.0, 0.0, -10.0, 500_000.0);
+    ///
+    /// let a: f64 = transform["a"];
+    /// let b: f64 = transform["b"];
+    /// let c: f64 = transform["xoff"];
+    /// let d: f64 = transform["d"];
+    /// let e: f64 = transform["e"];
+    /// let f: f64 = transform["yoff"];
+    /// ```
+    fn index(&self, idx: &str) -> &Self::Output {
+        match idx {
+            "a" => &self.0[0][0],
+            "b" => &self.0[0][1],
+            "xoff" => &self.0[0][2],
+            "d" => &self.0[1][0],
+            "e" => &self.0[1][1],
+            "yoff" => &self.0[1][2],
+            _ => &f64::NAN,
```

Comment:
Not ideal to return NaN here, would prefer if the possible indexes are limited to just 6 possible values, maybe with an enum?

Reply:
With getters this wouldn't be an issue.

Reply:
Out-of-bounds indexing in Rust is usually a panic, right? I'm not sure if this is strictly out of bounds, though.

Reply:
Yeah on second thought, getter methods might actually be better. I've refactored things in commit a0fdea520dbd501cc177960fe824e1450676f613